### PR TITLE
Gui: Prevent crash when calling stderr.write and stdout.write without arguments

### PR DESCRIPTION
**Pull request by marioalexis84**
Steps to reproduce the crash:
import sys
sys.stderr.write()
#same using sys.stdout.write()

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [x]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [x]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

File: src/Gui/PythonConsolePy.cpp [hunk 1]
```diff
@@ -78,9 +78,11 @@ Py::Object PythonStdout::repr()
 Py::Object PythonStdout::write(const Py::Tuple& args)
 {
     try {
-        Py::Object output(args[0]);
-        if (PyUnicode_Check(output.ptr())) {
-            PyObject* unicode = PyUnicode_AsEncodedString(output.ptr(), "utf-8", 0);
+        PyObject* output;
+        if (!PyArg_ParseTuple(args.ptr(), "O", &output))
+            throw Py::Exception();
```
Comment (AjinkyaDahale) [inline src/Gui/PythonConsolePy.cpp:83]: Could this maybe have some message attached to it? With these changes `sys.stderr.write()` returns nothing.

**Comment by AjinkyaDahale** [comment]
All tests appear to pass on my Ubuntu MATE 20.04 machine and there's no seg fault on the commands. However, there's no further notification that. An "out of bound" or "insufficient arguments" exception should be raised instead.

**Comment by marioalexis84** [comment]
I modified the code to check the argument type. If the type is not str or there is no argument, a TypeError exception is thrown with the message "* .write() takes exactly one argument of type str".
The "else" block is not necessary since Python's str type is already PyUnicode_Type

**Comment by berndhahnebach** [comment]
Following a link to the branch on the CI-repository:

https://gitlab.com/berndhahnebach/FreeCAD/-/commits/PR_5060

The CI-status is available on the latest commit of the branch.
If there is no status available the PR should be rebased on latest master.
Check pipeline branches to see if your PR has been run by the CI.

https://gitlab.com/berndhahnebach/FreeCAD/-/pipelines?scope=branches

**Comment by berndhahnebach** [comment]
FreeCAD CI-repository:
branch:
https://gitlab.com/berndhahnebach/FreeCAD/-/commits/PR_4884

pipeline:
https://gitlab.com/berndhahnebach/FreeCAD/-/pipelines/376846317

status:
success

all pipelines for each branch: 
https://gitlab.com/berndhahnebach/FreeCAD/-/pipelines?scope=branches

**Comment by donovaly** [comment]
@AjinkyaDahale Is this PR OK in your opinion to be merged?

**Comment by AjinkyaDahale** [comment]
@donovaly It has been quite some time since I checked this PR, but it does appear that the author fixed whatever comments I had on it. 

I think it might be good to go apart from one detail, viz. a `try...catch` statement has been replaced with an `if` check for one case (which is a seg fault). So if some other exception happens the behavior will be different than expected. If you believe that's fine I don't have any further issues.

**Comment by donovaly** [comment]
> a try...catch statement has been replaced with an if check

@marioalexis84 , can you please have a look at this? I cannot see a reason why a try-catch clause should be removed.

**Comment by donovaly** [comment]
OK, I understand. -> merging.

Do you think it is worth to be backported to FC 0.19?

**Comment by marioalexis84** [comment]
I do not think that it's necessary

**Comment by donovaly** [comment]
OK, thanks.

**Pull request merged**